app/services/discover/register.rb

```diff
@@ -0,0 +1,34 @@
+module Discover
+  class Register
+    FOREM_DISCOVER_URL = "https://discover.forem.com/api/forems/register".freeze
+
+    def initialize(domain:)
+      @domain = domain
+    end
+
+    def self.call(**args)
+      new(**args).call
+    end
+
+    def call
+      return unless @domain
```

Comment:
**Non-blocking suggestion:** I'm guessing `Settings::General.app_domain` is usually empty for local development environments, but I personally use ngrok quite often and have that config enabled. So to avoid "false registrations" it's maybe worth adding a production environment check here?

It may not be a common thing, because AFAICT this would only happen when working with admin onboarding flow, but maybe something like this would likely avoid it happening altogether?

```ruby
return unless @domain && Rails.env.production?
```

Reply:
@fdoxyz that's a good call. I'm typically hesitant because environment gates make things difficult to test. I updated it to not do it in development. Since we mock the requests in test, it should be okay there.